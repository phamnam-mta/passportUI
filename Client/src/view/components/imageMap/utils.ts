import { Feature } from "ol";
import { Extent } from "ol/extent";
import Polygon from "ol/geom/Polygon";
import ImageLayer from "ol/layer/Image";
import Layer from "ol/layer/Layer";
import VectorLayer from "ol/layer/Vector";
import Projection from "ol/proj/Projection";
import ImageSource from "ol/source/ImageStatic";
import VectorSource from "ol/source/Vector";
import { LAYER_NAME } from "./constants";
import { IRegion } from "./contracts";

export type Dimension = {
    width: number;
    height: number;
};

export const degreeToRadians = (degree: number) => {
    // convert degree to radians
    return (degree * Math.PI * 2) / 360;
};

export const makeLayerFilter = (layerName) => {
    return {
        layerFilter: (layer: Layer) => layer.get(LAYER_NAME) === layerName,
    };
};

export const makeVectorLayer = (layerName, options = {}): Layer => {
    return new VectorLayer({
        [LAYER_NAME]: layerName,
        source: new VectorSource(),
        ...options,
    } as any);
};

export const makeImageLayer = (layerName, uri, projection, imageExtent, options = {}): Layer => {
    return new ImageLayer({
        [LAYER_NAME]: layerName,
        source: makeImageSource(uri, projection, imageExtent),
        ...options,
    } as any);
};

export const makeImageSource = (url: string, projection: Projection, imageExtent: Extent) => {
    return new ImageSource({ url, projection, imageExtent });
};

export const convertRegionToFeature = (region: IRegion, imageExtent: Extent, isOcrProposal: boolean = false) => {
    const coordinates: Array<[number, number]> = [];
    const boundingBox = region.id.split(",").map(parseFloat);
    const imageWidth = imageExtent[2] - imageExtent[0];
    const imageHeight = imageExtent[3] - imageExtent[1];
    for (let i = 0; i < boundingBox.length; i += 2) {
        coordinates.push([Math.round(boundingBox[i] * imageWidth), Math.round((1 - boundingBox[i + 1]) * imageHeight)]);
    }
    const feature = new Feature({
        geometry: new Polygon([coordinates]),
    });
    feature.setProperties({
        id: region.id,
        text: region.value,
        highlighted: false,
        isOcrProposal,
    });
    feature.setId(region.id);

    return feature;
};

export const createRegionIdFromPolygon = (polygon: number[], page: number): string => {
    return polygon.join(",") + ":" + page;
};

export const convertToImageMapCoordinates = (polygon: number[], canvasSize: Dimension, documentSize: Dimension) => {
    const { width: canvasWidth, height: canvasHeight } = canvasSize;
    const { width: ocrWidth, height: ocrHeight } = documentSize;

    const coordinates: number[][] = [];
    for (let i = 0; i < polygon.length; i += 2) {
        coordinates.push([(polygon[i] / ocrWidth) * canvasWidth, (1 - polygon[i + 1] / ocrHeight) * canvasHeight]);
    }
    return coordinates;
};

export const getBoundingBoxFromFeatureId = (id) => id.split(",").map(parseFloat);
