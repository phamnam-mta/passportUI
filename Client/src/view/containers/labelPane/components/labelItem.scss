@import "~@fluentui/react/dist/sass/References";

.label-item {
    padding: 5px 8px;
    color: $ms-color-gray130;

    &:hover {
        cursor: pointer;
        background-color: $ms-color-gray20;
    }

    .label-item-entry {
        display: grid;
        grid-template-columns: 24px 24px 1fr 24px;
        align-items: center;
    }

    .label-item-result {
        display: grid;
        grid-template-columns: 24px 1fr 24px;
        align-items: center;
    }

    .label-item-value {
        padding: 6px 0;
        cursor: pointer;

        .label-item-text {
            font-size: 14px;
            font-weight: 600;
        }
    }

    .label-item-icon {
        font-size: 16px;
    }

    .no-select {
        -webkit-touch-callout: none; /* iOS Safari */
        -webkit-user-select: none; /* Safari */
        -khtml-user-select: none; /* Konqueror HTML */
        -moz-user-select: none; /* Old versions of Firefox */
        -ms-user-select: none; /* Internet Explorer/Edge */
        user-select: none; /* Non-prefixed version, currently
                                    supported by Chrome, Edge, Opera and Firefox */
    }
}
