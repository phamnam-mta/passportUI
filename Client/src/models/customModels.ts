import { FeatureCategory } from "view/components/imageMap/contracts";
import { Polygon } from "./analyzeResult";

export type CustomModel = {
    modelId: string;
    description: string;
    createdDateTime: string;
    apiVersion?: string;
    tags?: {
        modelType?: string;
        [key: string]: string | undefined;
    };
};

export type PrimitiveField = {
    fieldKey: string;
    fieldType: FieldType;
    fieldFormat: FieldFormat;
};

export type ObjectField = {
    fieldKey: string;
    fieldType: FieldType;
    fields: Field[];
    fieldFormat: FieldFormat;
    visualizationHint?: VisualizationHint;
};

export type ArrayField = {
    fieldKey: string;
    fieldType: FieldType;
    itemType: string;
    visualizationHint?: VisualizationHint;
};

export type Field = PrimitiveField | ObjectField | ArrayField;

export type FieldsWithOrder = Field & { order: number };

export enum FieldType {
    String = "string",
    Number = "number",
    Date = "date",
    Time = "time",
    Integer = "integer",
    SelectionMark = "selectionMark",
    Signature = "signature",
    Array = "array",
    Object = "object",
}

export enum FieldFormat {
    NotSpecified = "not-specified",
    Currency = "currency",
    Decimal = "decimal",
    DecimalCommaSeparated = "decimal-comma-separated",
    NoWhiteSpaces = "no-white-spaces",
    Alphanumeric = "alphanumeric",
    DMY = "dmy",
    MDY = "mdy",
    YMD = "ymd",
}

export enum VisualizationHint {
    Horizontal = "horizontal",
    Vertical = "vertical",
}

export type Labels = {
    [documentName: string]: Label[];
};

export type Label = {
    label: string;
    value: LabelValue[];
    labelType?: LabelType;
};

export type LabelValue = {
    boundingBoxes: Polygon[];
    page: number;
    text: string;
};

export type LabelValueCandidate = {
    boundingBoxes: Polygon[];
    page: number;
    text: string;
    category: FeatureCategory;
    alreadyAssignedLabelName?: string;
};

export enum LabelType {
    Words = "words",
    Region = "region",
}

export type Definitions = {
    [objectName: string]: ObjectField;
};

export enum TableType {
    dynamic = "dynamic",
    fixed = "fixed",
}

export enum HeaderType {
    row = "row",
    column = "column",
}
