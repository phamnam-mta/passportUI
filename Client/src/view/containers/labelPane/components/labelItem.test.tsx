import * as React from "react";
import { act } from "react-dom/test-utils";

import { flushPromises } from "utils/test";
import { Field, FieldFormat, FieldType, Label, LabelType } from "models/customModels";
import { LabelItem } from "./labelItem";
import { encodeLabelString } from "utils/customModel";
import { shallow } from "enzyme";

jest.spyOn(global, "setTimeout");

describe("<LabelItem />", () => {
    let baseProps;
    const mockStringField: Field = {
        fieldKey: "mockField",
        fieldType: FieldType.String,
        fieldFormat: FieldFormat.NotSpecified,
    };
    const mockStringLabel: Label = {
        label: mockStringField.fieldKey,
        value: [{ page: 1, boundingBoxes: [], text: "abc" }],
    };
    const mockStringRegionLabel: Label = {
        label: mockStringField.fieldKey,
        labelType: LabelType.Region,
        value: [{ page: 1, boundingBoxes: [], text: "" }],
    };
    const mockSignatureField: Field = {
        fieldKey: "mockSignature",
        fieldType: FieldType.Signature,
        fieldFormat: FieldFormat.NotSpecified,
    };
    const mockSignatureLabel: Label = {
        label: mockSignatureField.fieldKey,
        labelType: LabelType.Region,
        value: [{ page: 1, boundingBoxes: [], text: "" }],
    };
    const mockSelectionMarkField: Field = {
        fieldKey: "mockSelectionMark",
        fieldType: FieldType.SelectionMark,
        fieldFormat: FieldFormat.NotSpecified,
    };
    const mockSelectionMarkLabel: Label = {
        label: mockSelectionMarkField.fieldKey,
        value: [{ page: 1, boundingBoxes: [], text: "selected" }],
    };
    const mockTableField: Field = {
        fieldKey: "mockTable",
        fieldType: FieldType.Array,
        itemType: "mockTable_object",
    };
    const mockTableLabel: Label = {
        label: mockTableField.fieldKey,
        value: [{ page: 1, boundingBoxes: [], text: "" }],
    };

    beforeEach(() => {
        baseProps = {
            field: mockStringField,
            color: "purple",
            disableItemOperations: false,
            setColorForFieldsByName: jest.fn(),
            onSwitchSubType: jest.fn(),
            onDeleteField: jest.fn(),
            onDeleteLabel: jest.fn(),
            onRenameField: jest.fn(),
            onGetRenameErrorMessage: jest.fn(),
            onClickTableField: jest.fn(),
            onClickField: jest.fn(),
        };
    });

    describe("Rendering", () => {
        it("should match snapshot, when there's no label", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it's renaming", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);
            wrapper.setState({ isRenaming: true });

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it's string field", () => {
            const props = { ...baseProps, label: mockStringLabel };

            const wrapper = shallow(<LabelItem {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it's string region field", () => {
            const props = { ...baseProps, label: mockStringRegionLabel };

            const wrapper = shallow(<LabelItem {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it's signature field", () => {
            const props = { ...baseProps, field: mockSignatureField, label: mockSignatureLabel };

            const wrapper = shallow(<LabelItem {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it's selectionMark field", () => {
            const props = { ...baseProps, field: mockSelectionMarkField, label: mockSelectionMarkLabel };

            const wrapper = shallow(<LabelItem {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it's table field", () => {
            const props = { ...baseProps, field: mockTableField, label: mockTableLabel };

            const wrapper = shallow(<LabelItem {...props} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Event Handling", () => {
        it("should delete label, when delete button is clicked", () => {
            const props = { ...baseProps, label: mockStringLabel };

            const wrapper = shallow(<LabelItem {...props} />);
            const button = wrapper.find("DeleteButton");
            const mockEvent = { stopPropagation: jest.fn() };
            button.simulate("click", mockEvent as any);

            expect(props.onDeleteLabel).toBeCalledTimes(1);
            expect(props.onDeleteLabel).toBeCalledWith(props.field.fieldKey);
            expect(mockEvent.stopPropagation).toBeCalledTimes(1);
        });

        it("should handle modal close, when modal onClose is triggered", async () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);
            wrapper.setState({
                isConfirmModalOpen: true,
                confirmOperation: "delete",
            });
            const modal = wrapper.find("MessageModal") as any;
            modal.prop("onClose")();

            expect(wrapper.state("isConfirmModalOpen")).toBe(false);
        });

        // Switch sub type.
        it("should switch sub type, when menu button onSwitchSubType is triggered", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);
            const button = wrapper.find("MenuButton") as any;
            button.prop("onSwitchSubType")(FieldType.Number);

            expect(baseProps.onSwitchSubType).toBeCalledTimes(1);
            expect(baseProps.onSwitchSubType).toBeCalledWith(baseProps.field.fieldKey, FieldType.Number);
        });

        // Delete field.
        it("should set modal open and confirm type, when menu button onDeleteField is triggered", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);
            const button = wrapper.find("MenuButton") as any;
            button.prop("onDeleteField")();

            expect(wrapper.state("isConfirmModalOpen")).toBe(true);
            expect(wrapper.state("confirmOperation")).toBe("delete");
        });

        it("should handle delete field, when confirm to delete field", async () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);
            wrapper.setState({
                isConfirmModalOpen: true,
                confirmOperation: "delete",
            });
            const modal = wrapper.find("MessageModal") as any;
            modal.prop("onActionButtonClick")();

            expect(wrapper.state("isConfirmModalLoading")).toBe(true);
            expect(baseProps.onDeleteField).toBeCalledTimes(1);
            expect(baseProps.onDeleteField).toBeCalledWith(baseProps.field.fieldKey);
            await flushPromises();
            expect(wrapper.state("isConfirmModalOpen")).toBe(false);
            expect(wrapper.state("isConfirmModalLoading")).toBe(false);
            expect(wrapper.state("confirmOperation")).toBeUndefined();
        });

        it("should handle delete field, when pressing Enter key to confirm deletion", async () => {
            const map: any = {};
            const mockAddEventListener = jest.fn((event, callback) => {
                map[event] = callback;
            });
            const mockRemoveEventListener = jest.fn();
            document.addEventListener = mockAddEventListener;
            document.removeEventListener = mockRemoveEventListener;

            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            const button = wrapper.find("MenuButton") as any;
            button.prop("onDeleteField")();
            const onKeyDown = wrapper.instance().onKeyDown;
            expect(mockAddEventListener).toBeCalledWith("keydown", onKeyDown, true);
            act(() =>
                map.keydown({
                    key: "Enter",
                })
            );
            expect(wrapper.state("isConfirmModalLoading")).toBe(true);
            expect(baseProps.onDeleteField).toBeCalledTimes(1);
            expect(baseProps.onDeleteField).toBeCalledWith(baseProps.field.fieldKey);
            await flushPromises();
            expect(wrapper.state("isConfirmModalOpen")).toBe(false);
            expect(wrapper.state("isConfirmModalLoading")).toBe(false);
            expect(wrapper.state("confirmOperation")).toBeUndefined();
            expect(mockRemoveEventListener).toBeCalledWith("keydown", onKeyDown, true);
        });

        // Rename field.
        it("should set isRenaming, when menu button is triggered", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);
            const button = wrapper.find("MenuButton") as any;
            button.prop("onRenameField")();

            expect(wrapper.state("isRenaming")).toBe(true);
        });

        it("should handle rename dismiss, when rename text field onBlur is triggered", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />);
            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("blur");

            expect(wrapper.state("isRenaming")).toBe(false);
        });

        it("should not call onClickTableField, when click on field rename textField", async () => {
            const mockEvent = { stopPropagation: jest.fn() };
            const wrapper = shallow(<LabelItem {...baseProps} />);
            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("click", mockEvent);

            expect(mockEvent.stopPropagation).toBeCalledTimes(1);
            expect(baseProps.onClickTableField).not.toBeCalled();
        });

        it("should not handle rename dismiss, when rename text field onBlur is triggered and isEnteringRename", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            wrapper.setState({ isRenaming: true });
            wrapper.instance().isEnteringRename = true;
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("blur");

            expect(wrapper.state("isRenaming")).toBe(true);
        });

        it("should open rename modal, when pressing enter with a new name", () => {
            const value = "testRename";
            const mockEvent = { key: "Enter", target: { value }, stopPropagation: jest.fn() };

            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("keydown", mockEvent);

            expect(mockEvent.stopPropagation).toBeCalledTimes(1);
            expect(wrapper.state("isConfirmModalOpen")).toBe(true);
            expect(wrapper.state("confirmOperation")).toBe("rename");
            expect(wrapper.state("newFieldName")).toBe(value);
            expect(wrapper.instance().isEnteringRename).toBe(true);
        });

        it("should not open rename modal, when pressing enter with an invalid name", () => {
            const props = {
                ...baseProps,
                onGetRenameErrorMessage: jest.fn().mockReturnValue("error"),
            };
            const value = "testRename";
            const mockEvent = { key: "Enter", target: { value }, stopPropagation: jest.fn() };
            const wrapper = shallow(<LabelItem {...props} />) as any;

            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("keydown", mockEvent);

            expect(mockEvent.stopPropagation).toBeCalledTimes(1);
            expect(wrapper.state("isConfirmModalOpen")).toBe(false);
        });

        it.each([
            ["the same name", "mockField"],
            ["empty name", ""],
        ])("should handle rename dismiss, when pressing enter with %s", (_, value) => {
            const mockEvent = { key: "Enter", target: { value }, stopPropagation: jest.fn() };
            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("keydown", mockEvent);

            expect(mockEvent.stopPropagation).toBeCalledTimes(1);
            expect(wrapper.state("isRenaming")).toBe(false);
        });

        it("should handle rename dismiss, when pressing Escape", () => {
            const mockEvent = { key: "Escape", target: { value: "" }, stopPropagation: jest.fn() };
            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("keydown", mockEvent);

            expect(mockEvent.stopPropagation).toBeCalledTimes(1);
            expect(wrapper.state("isRenaming")).toBe(false);
        });

        it("should handle rename field, when confirm to rename field", async () => {
            const newFieldName = "newFieldName";

            const wrapper = shallow(<LabelItem {...baseProps} />);
            wrapper.setState({
                isConfirmModalOpen: true,
                confirmOperation: "rename",
                newFieldName,
            });
            const modal = wrapper.find("MessageModal") as any;
            modal.prop("onActionButtonClick")();

            expect(wrapper.state("isConfirmModalLoading")).toBe(true);
            expect(baseProps.onRenameField).toBeCalledTimes(1);
            expect(baseProps.onRenameField).toBeCalledWith(baseProps.field.fieldKey, newFieldName);
            await flushPromises();
            expect(wrapper.state("isConfirmModalOpen")).toBe(false);
            expect(wrapper.state("isConfirmModalLoading")).toBe(false);
            expect(wrapper.state("confirmOperation")).toBeUndefined();
            expect(wrapper.state("newFieldName")).toBeUndefined();
        });

        it("should handle rename field, when pressing Enter key to confirm rename", async () => {
            const value = "testRename";
            const mockEvent = { key: "Enter", target: { value }, stopPropagation: jest.fn() };
            const map: any = {};
            const mockAddEventListener = jest.fn((event, callback) => {
                map[event] = callback;
            });
            const mockRemoveEventListener = jest.fn();
            document.addEventListener = mockAddEventListener;
            document.removeEventListener = mockRemoveEventListener;

            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            const onKeyDown = wrapper.instance().onKeyDown;
            const button = wrapper.find("MenuButton") as any;
            button.prop("onRenameField")();
            const textField = wrapper.find("StyledTextFieldBase");
            textField.simulate("keydown", mockEvent);

            expect(setTimeout).toHaveBeenCalledTimes(1);
            expect(mockEvent.stopPropagation).toBeCalledTimes(1);
            expect(mockAddEventListener).toBeCalledWith("keydown", onKeyDown, true);
            act(() =>
                map.keydown({
                    key: "Enter",
                })
            );
            expect(wrapper.state("isConfirmModalOpen")).toBe(true);
            expect(wrapper.state("confirmOperation")).toBe("rename");
            expect(wrapper.state("newFieldName")).toBe(value);
            expect(wrapper.instance().isEnteringRename).toBe(true);
            expect(wrapper.state("isConfirmModalLoading")).toBe(true);
            expect(baseProps.onRenameField).toBeCalledTimes(1);
            expect(baseProps.onRenameField).toBeCalledWith(baseProps.field.fieldKey, value);
            await flushPromises();
            expect(wrapper.state("isConfirmModalOpen")).toBe(false);
            expect(wrapper.state("isConfirmModalLoading")).toBe(false);
            expect(wrapper.state("confirmOperation")).toBeUndefined();
            expect(wrapper.state("newFieldName")).toBeUndefined();
            expect(mockRemoveEventListener).toBeCalledWith("keydown", onKeyDown, true);
        });

        it("should handle table field click, when table field is clicked", () => {
            const props = { ...baseProps, field: mockTableField, label: mockTableLabel };

            const wrapper = shallow(<LabelItem {...props} />);
            const item = wrapper.find(".label-item");
            item.simulate("click");

            expect(baseProps.onClickTableField).toBeCalledTimes(1);
            expect(baseProps.onClickTableField).toBeCalledWith(mockTableField);
            expect(baseProps.onClickField).not.toBeCalled();
        });

        it("should handle field click, when non-table field is clicked", () => {
            let wrapper = shallow(<LabelItem {...baseProps} />);
            let item = wrapper.find(".label-item");
            item.simulate("click");

            expect(baseProps.onClickField).toBeCalledTimes(1);
            expect(baseProps.onClickField).toBeCalledWith(encodeLabelString(mockStringField.fieldKey));

            let props = { ...baseProps, field: mockSelectionMarkField, label: mockSelectionMarkLabel };
            wrapper = shallow(<LabelItem {...props} />);
            item = wrapper.find(".label-item");
            item.simulate("click");

            expect(baseProps.onClickField).toBeCalledTimes(2);
            expect(baseProps.onClickField).toBeCalledWith(encodeLabelString(mockSelectionMarkField.fieldKey));

            props = { ...baseProps, field: mockSignatureField, label: mockSignatureLabel };
            wrapper = shallow(<LabelItem {...props} />);
            item = wrapper.find(".label-item");
            item.simulate("click");

            expect(baseProps.onClickField).toBeCalledTimes(3);
            expect(baseProps.onClickField).toBeCalledWith(encodeLabelString(mockSignatureField.fieldKey));

            expect(baseProps.onClickTableField).not.toBeCalled();
        });
    });

    describe("Error Handling", () => {
        it("should return error message, when name is different", () => {
            const name = "testRename";

            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");
            textField.prop("onGetErrorMessage")(name);

            expect(baseProps.onGetRenameErrorMessage).toBeCalledTimes(1);
            expect(baseProps.onGetRenameErrorMessage).toBeCalledWith(name);
        });

        it("should not return error message, when name is the same", () => {
            const wrapper = shallow(<LabelItem {...baseProps} />) as any;
            wrapper.setState({ isRenaming: true });
            const textField = wrapper.find("StyledTextFieldBase");

            expect(textField.prop("onGetErrorMessage")(baseProps.field.fieldKey)).toBeUndefined();
            expect(baseProps.onGetRenameErrorMessage).not.toBeCalled();
        });
    });
});
