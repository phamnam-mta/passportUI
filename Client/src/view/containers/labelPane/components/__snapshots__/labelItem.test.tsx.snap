// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelItem /> Rendering should match snapshot, when it's renaming 1`] = `
<Stack
  className="label-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="label-item-entry no-select"
  >
    <div>
      <Icon
        iconName="GlobalNavButton"
      />
    </div>
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "purple",
          "fontSize": 16,
        }
      }
    />
    <StyledTextFieldBase
      autoComplete="off"
      componentRef={
        Object {
          "current": null,
        }
      }
      defaultValue="mockField"
      id="rename-textfield"
      onBlur={[Function]}
      onClick={[Function]}
      onGetErrorMessage={[Function]}
      onKeyDown={[Function]}
      styles={
        Object {
          "fieldGroup": Object {
            "height": 24,
          },
        }
      }
    />
    <MenuButton
      hasSubType={true}
      onDeleteField={[Function]}
      onRenameField={[Function]}
      onSwitchSubType={[Function]}
      subType="string"
    />
    <MessageModal
      actionButtonText="Yes"
      body={
        <Text
          variant="medium"
        >
          <Text>
            Are you sure you want to rename 
            <b>
              mockField
            </b>
             to 
            <b />
            ? All labels and regions assigned to this field will be changed thoroughly
          </Text>
        </Text>
      }
      isLoading={false}
      isOpen={false}
      onActionButtonClick={[Function]}
      onClose={[Function]}
      onDismiss={[Function]}
      rejectButtonText="No"
      title="Rename Field"
    />
  </div>
</Stack>
`;

exports[`<LabelItem /> Rendering should match snapshot, when it's selectionMark field 1`] = `
<Stack
  className="label-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="label-item-entry no-select"
  >
    <div>
      <Icon
        iconName="GlobalNavButton"
      />
    </div>
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "purple",
          "fontSize": 16,
        }
      }
    />
    <Text
      variant="medium"
    >
      mockSelectionMark
    </Text>
    <MenuButton
      hasSubType={false}
      onDeleteField={[Function]}
      onRenameField={[Function]}
      onSwitchSubType={[Function]}
      subType="selectionMark"
    />
    <MessageModal
      actionButtonText="Yes"
      body={
        <Text
          variant="medium"
        >
          <Text>
            Are you sure you want to rename 
            <b>
              mockSelectionMark
            </b>
             to 
            <b />
            ? All labels and regions assigned to this field will be changed thoroughly
          </Text>
        </Text>
      }
      isLoading={false}
      isOpen={false}
      onActionButtonClick={[Function]}
      onClose={[Function]}
      onDismiss={[Function]}
      rejectButtonText="No"
      title="Rename Field"
    />
  </div>
  <div
    className="label-item-result label-item-value"
  >
    <div />
    <Stack
      horizontal={true}
      tokens={
        Object {
          "childrenGap": 6,
        }
      }
      verticalAlign="center"
    >
      <FontIcon
        className="label-item-icon"
        iconName="CheckboxComposite"
      />
      <Text
        className="label-item-text"
      >
        selected
      </Text>
    </Stack>
    <DeleteButton
      onClick={[Function]}
    />
  </div>
</Stack>
`;

exports[`<LabelItem /> Rendering should match snapshot, when it's signature field 1`] = `
<Stack
  className="label-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="label-item-entry no-select"
  >
    <div>
      <Icon
        iconName="GlobalNavButton"
      />
    </div>
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "purple",
          "fontSize": 16,
        }
      }
    />
    <Text
      variant="medium"
    >
      mockSignature
    </Text>
    <MenuButton
      hasSubType={false}
      onDeleteField={[Function]}
      onRenameField={[Function]}
      onSwitchSubType={[Function]}
      subType="signature"
    />
    <MessageModal
      actionButtonText="Yes"
      body={
        <Text
          variant="medium"
        >
          <Text>
            Are you sure you want to rename 
            <b>
              mockSignature
            </b>
             to 
            <b />
            ? All labels and regions assigned to this field will be changed thoroughly
          </Text>
        </Text>
      }
      isLoading={false}
      isOpen={false}
      onActionButtonClick={[Function]}
      onClose={[Function]}
      onDismiss={[Function]}
      rejectButtonText="No"
      title="Rename Field"
    />
  </div>
  <div
    className="label-item-result label-item-value"
  >
    <div />
    <FontIcon
      className="label-item-icon"
      iconName="WhiteBoardApp16"
    />
    <DeleteButton
      onClick={[Function]}
    />
  </div>
</Stack>
`;

exports[`<LabelItem /> Rendering should match snapshot, when it's string field 1`] = `
<Stack
  className="label-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="label-item-entry no-select"
  >
    <div>
      <Icon
        iconName="GlobalNavButton"
      />
    </div>
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "purple",
          "fontSize": 16,
        }
      }
    />
    <Text
      variant="medium"
    >
      mockField
    </Text>
    <MenuButton
      hasSubType={true}
      onDeleteField={[Function]}
      onRenameField={[Function]}
      onSwitchSubType={[Function]}
      subType="string"
    />
    <MessageModal
      actionButtonText="Yes"
      body={
        <Text
          variant="medium"
        >
          <Text>
            Are you sure you want to rename 
            <b>
              mockField
            </b>
             to 
            <b />
            ? All labels and regions assigned to this field will be changed thoroughly
          </Text>
        </Text>
      }
      isLoading={false}
      isOpen={false}
      onActionButtonClick={[Function]}
      onClose={[Function]}
      onDismiss={[Function]}
      rejectButtonText="No"
      title="Rename Field"
    />
  </div>
  <div
    className="label-item-result label-item-value"
  >
    <div />
    <Text
      className="label-item-text"
    >
      abc
    </Text>
    <DeleteButton
      onClick={[Function]}
    />
  </div>
</Stack>
`;

exports[`<LabelItem /> Rendering should match snapshot, when it's string region field 1`] = `
<Stack
  className="label-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="label-item-entry no-select"
  >
    <div>
      <Icon
        iconName="GlobalNavButton"
      />
    </div>
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "purple",
          "fontSize": 16,
        }
      }
    />
    <Text
      variant="medium"
    >
      mockField
    </Text>
    <MenuButton
      hasSubType={true}
      onDeleteField={[Function]}
      onRenameField={[Function]}
      onSwitchSubType={[Function]}
      subType="string"
    />
    <MessageModal
      actionButtonText="Yes"
      body={
        <Text
          variant="medium"
        >
          <Text>
            Are you sure you want to rename 
            <b>
              mockField
            </b>
             to 
            <b />
            ? All labels and regions assigned to this field will be changed thoroughly
          </Text>
        </Text>
      }
      isLoading={false}
      isOpen={false}
      onActionButtonClick={[Function]}
      onClose={[Function]}
      onDismiss={[Function]}
      rejectButtonText="No"
      title="Rename Field"
    />
  </div>
  <div
    className="label-item-result label-item-value"
  >
    <div />
    <FontIcon
      className="label-item-icon"
      iconName="SingleColumnEdit"
    />
    <DeleteButton
      onClick={[Function]}
    />
  </div>
</Stack>
`;

exports[`<LabelItem /> Rendering should match snapshot, when it's table field 1`] = `
<Stack
  className="label-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="label-item-entry no-select"
  >
    <div>
      <Icon
        iconName="GlobalNavButton"
      />
    </div>
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "purple",
          "fontSize": 16,
        }
      }
    />
    <Text
      variant="medium"
    >
      mockTable
    </Text>
    <MenuButton
      hasSubType={false}
      onDeleteField={[Function]}
      onRenameField={[Function]}
      onSwitchSubType={[Function]}
      subType="array"
    />
    <MessageModal
      actionButtonText="Yes"
      body={
        <Text
          variant="medium"
        >
          <Text>
            Are you sure you want to rename 
            <b>
              mockTable
            </b>
             to 
            <b />
            ? All labels and regions assigned to this field will be changed thoroughly
          </Text>
        </Text>
      }
      isLoading={false}
      isOpen={false}
      onActionButtonClick={[Function]}
      onClose={[Function]}
      onDismiss={[Function]}
      rejectButtonText="No"
      title="Rename Field"
    />
  </div>
  <div
    className="label-item-result label-item-value"
  >
    <div />
    <FontIcon
      className="label-item-icon"
      iconName="Table"
    />
    <DeleteButton
      onClick={[Function]}
    />
  </div>
</Stack>
`;

exports[`<LabelItem /> Rendering should match snapshot, when there's no label 1`] = `
<Stack
  className="label-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="label-item-entry no-select"
  >
    <div>
      <Icon
        iconName="GlobalNavButton"
      />
    </div>
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "purple",
          "fontSize": 16,
        }
      }
    />
    <Text
      variant="medium"
    >
      mockField
    </Text>
    <MenuButton
      hasSubType={true}
      onDeleteField={[Function]}
      onRenameField={[Function]}
      onSwitchSubType={[Function]}
      subType="string"
    />
    <MessageModal
      actionButtonText="Yes"
      body={
        <Text
          variant="medium"
        >
          <Text>
            Are you sure you want to rename 
            <b>
              mockField
            </b>
             to 
            <b />
            ? All labels and regions assigned to this field will be changed thoroughly
          </Text>
        </Text>
      }
      isLoading={false}
      isOpen={false}
      onActionButtonClick={[Function]}
      onClose={[Function]}
      onDismiss={[Function]}
      rejectButtonText="No"
      title="Rename Field"
    />
  </div>
</Stack>
`;
